libs/api/repository/src/lib/gn4/gn4.provider.ts

```diff
@@ -0,0 +1,14 @@
+import { Provider } from '@angular/core'
+import { PlatformServiceInterface } from '@geonetwork-ui/common/domain/platform.service.interface'
+import { Gn4PlatformService } from './platform/gn4-platform.service'
+import { Gn4PlatformMapper } from './platform/gn4-platform.mapper'
+
+export function provideGn4(): Provider[] {
+  return [
+    {
+      provide: PlatformServiceInterface,
+      useClass: Gn4PlatformService,
+    },
+    Gn4PlatformMapper,
```

Comment:
I don't see why this should be provided? Isn't it an explicit dependency of Gn4PlatformService?

Reply:
Because it's not injected in root, so it has to be injected somehow.
There is no `providers` option from services.